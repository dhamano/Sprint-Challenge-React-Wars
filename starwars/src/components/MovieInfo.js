import React, { useState, useEffect } from 'react';
import { getFromStarWarsAPI } from '../services/fetchData';
import Loader from 'react-loader-spinner';
import styled from 'styled-components';

const MovieInfo = props => {
	const [ charFilms, setCharFilms ] = useState('');

  useEffect( () => {
		getFromStarWarsAPI(props.movieURL).then( json => setCharFilms(json));
	}, []);

	const MovieList = styled.li`
		dt {
			font-weight: bold;
		}

		.full {
			width: 100%;
		}
	`

	const LoaderSpinner = styled.div`
		width: 100vw;
		height: 100vh;
		display: flex;
		justify-content: center;
		align-items: center;
	`
	
	if (!charFilms) {
		return (
			<LoaderSpinner>
				<Loader type="Grid" color="#000000" height="100" width="100" />
			</LoaderSpinner>
		)
	}

	return(
		<MovieList>
			<h3>{charFilms.title}</h3>
			<dl>
				<dt>release date:</dt>
				<dd>{charFilms.release_date}</dd>
				<dt>director:</dt>
				<dd>{charFilms.director}</dd>
				<dt className="full">synopsis:</dt>
				<dd className="full">{charFilms.opening_crawl}</dd>
			</dl>
		</MovieList>
	)
}

export default MovieInfo;