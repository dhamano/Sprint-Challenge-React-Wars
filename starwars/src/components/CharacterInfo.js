import React from 'react';
import styled from 'styled-components';
import Loader from 'react-loader-spinner';

import MovieInfo from './MovieInfo';

const CharacterInfo = props => {
	const CharCard = styled.div`
		width: 30%;
		padding: 20px;
		margin: 20px auto;;
		background: rgba(0,0,0,0.5);
		text-align: left;
		border: solid 1px #000;
		border-radius: 8px;

		h1 {
			margin-top: 0;
		}

		dl {
			display: flex;
			flex-flow: row wrap;
		}

		dt, dd {
			width: 50%;
			margin: 0;
		}

		.movies {
			width: 100%;
			margin: 0;

			ul, li {
				margin: 0;
				padding: 0;
				list-style-type: none;
			}
		}
	`

	const LoaderSpinner = styled.div`
		width: 100vw;
		height: 100vh;
		display: flex;
		justify-content: center;
		align-items: center;
	`
	
	if (!props.charInfo) {
		return (
			<LoaderSpinner>
				<Loader type="Grid" color="#000000" height="100" width="100" />
			</LoaderSpinner>
		)
	}

	return (
		<CharCard>
			<h1>{props.charInfo.name}</h1>
			<dl>
				<dt>birth year:</dt>
				<dd>{props.charInfo.birth_year}</dd>
				<dt>height:</dt>
				<dd>{props.charInfo.height}</dd>
				<dt>weight:</dt>
				<dd>{props.charInfo.mass}{`${props.charInfo.mass === 'unknown' ? '' : ' kg'}`}</dd>
				<dt>hair color:</dt>
				<dd>{props.charInfo.hair_color}</dd>
				<dt>skin color:</dt>
				<dd>{props.charInfo.skin_color}</dd>
				<dt>films:</dt>
				<dd className="movies">
					<ul>
						{
							props.charInfo.films.map( (movieURL, i) => {
								return <MovieInfo movieURL={movieURL} key={'movie'+i} />
							})
						}
					</ul>
				</dd>
      </dl>
		</CharCard>
	)
}

export default CharacterInfo;