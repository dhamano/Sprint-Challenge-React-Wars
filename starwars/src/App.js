import React, { useState, useEffect } from 'react';
import Loader from 'react-loader-spinner';
import styled from 'styled-components';

import { getFromStarWarsAPI } from './services/fetchData';

import CharacterInfo from './components/CharacterInfo';

const App = () => {
	const [ baseSWData, setBaseSWData ] = useState(null);

  useEffect( () => {
		getFromStarWarsAPI().then( json => setBaseSWData(json));
	}, []);

	const LoaderSpinner = styled.div`
		width: 100vw;
		height: 100vh;
		display: flex;
		justify-content: center;
		align-items: center;
	`
	
	if (!baseSWData) {
		return (
			<LoaderSpinner>
				<Loader type="Grid" color="#000000" height="100" width="100" />
			</LoaderSpinner>
		)
	}
	
	console.log('baseSWData',baseSWData);

	const CharList = styled.div`
		width: calc(100vw - 0.875rem);
		margin-top: 50px;
	`

	const CharCardList = styled.div`
		width: calc(100% - 0.875rem);
		display: flex;
		flex-flow: row wrap;
		justify-content: space-between;
	`

  return (
		<CharList className="App">
			<h1 className="Header">React Wars</h1>
			<CharCardList>
				{
					baseSWData.results.map( (charInfo, i) => {
						return <CharacterInfo charInfo={charInfo} key={i} />
					})
				}
			</CharCardList>
		</CharList>
  );
}

export default App;
