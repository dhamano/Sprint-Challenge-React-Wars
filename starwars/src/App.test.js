import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import CharacterInfo from './components/CharacterInfo';
import MovieInfo from './components/MovieInfo';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('<App />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
	});
});

describe('<CharacterInfo />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<CharacterInfo />).toJSON();
    expect(tree).toMatchSnapshot();
	});
});

describe('<MovieInfo />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<MovieInfo />).toJSON();
    expect(tree).toMatchSnapshot();
	});
});