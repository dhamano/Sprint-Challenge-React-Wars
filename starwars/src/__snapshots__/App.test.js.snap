// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="sc-bwzfXH elMKDE"
>
  <div>
    <svg
      fill="#000000"
      height="100"
      viewBox="0 0 105 105"
      width="100"
    >
      <circle
        cx="12.5"
        cy="12.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="0s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="12.5"
        cy="52.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="100ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="12.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="300ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="52.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="600ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="12.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="800ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="52.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="400ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="12.5"
        cy="92.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="700ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="92.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="500ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="92.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="200ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
    </svg>
  </div>
</div>
`;

exports[`<CharacterInfo /> should match snapshot 1`] = `
<div
  className="sc-bxivhb cmSWOQ"
>
  <div>
    <svg
      fill="#000000"
      height="100"
      viewBox="0 0 105 105"
      width="100"
    >
      <circle
        cx="12.5"
        cy="12.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="0s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="12.5"
        cy="52.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="100ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="12.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="300ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="52.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="600ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="12.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="800ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="52.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="400ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="12.5"
        cy="92.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="700ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="92.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="500ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="92.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="200ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
    </svg>
  </div>
</div>
`;

exports[`<MovieInfo /> should match snapshot 1`] = `
<div
  className="sc-EHOje hNhuEW"
>
  <div>
    <svg
      fill="#000000"
      height="100"
      viewBox="0 0 105 105"
      width="100"
    >
      <circle
        cx="12.5"
        cy="12.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="0s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="12.5"
        cy="52.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="100ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="12.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="300ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="52.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="600ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="12.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="800ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="52.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="400ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="12.5"
        cy="92.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="700ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="92.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="500ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="92.5"
        r="12.5"
      >
        <animate
          attributeName="fill-opacity"
          begin="200ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
    </svg>
  </div>
</div>
`;
